libs/wingc/src/diagnostic.rs

```diff
@@ -1,65 +1,107 @@
 use colored::Colorize;
 use std::fmt::Display;
+use tree_sitter::Point;
 
 use lsp_types::{Position, Range};
-use tree_sitter::Point;
 
 use crate::debug;
 
 pub type FileId = String;
-pub type CharacterLocation = Point;
-pub type ByteIndex = usize;
-
 pub type Diagnostics = Vec<Diagnostic>;
-
 pub type DiagnosticResult<T> = Result<T, ()>;
 
-#[derive(Debug, PartialEq, Eq, Hash, Clone)]
-pub struct WingSpan {
-	pub start: CharacterLocation,
-	pub end: CharacterLocation,
-	pub start_byte: ByteIndex,
-	pub end_byte: ByteIndex,
-	pub file_id: FileId,
+/// Line and character location in a UTF8 Wing source file
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct WingPoint {
```

Comment:
WDYT of a name like `Location` or `Position`?

Reply:
I felt a similar dilemma with `Span`, where the type is used in several places. tree-sitter has one, lsp-types has one, and I expect us to run into another one whenever you use a new crate for beautiful diagnostics.

Normally I wouldn't name types with Wing___, but this one felt helpful to avoid ambiguity (or being forced to put namespaces)